projects/valence/index.js

```diff
@@ -1,27 +1,133 @@
+      staked: true,
+    },
+    {
+      address: "neutron1yj6m02det6qd6wdugtfytfryzcmt9z9apls0sf4ww07hqq54zexqt5xgaq", //mars
+      staked: false,
+    },
+  ]
+}
+
+// Configurations for querying value for liquidity tokens for specific DEXs
+const DEX_CONFIGS = {
+  astroport: {
+    // regex to match liquidity tokens  
+    regex: ":astroport:share",
+    // function to extract pool address from liquidity token address
+    poolAddressExtractor: (lpToken, chain) => lpToken.replace(`${chain}:factory:`,"").replace(":astroport:share",""),
+    // function to query value of liquidity token
+    // returns {amount: number, denom: string}
+    queryShareValue: async ({
+      poolAddress,
+      lpBalance,
+      chain
+    }) => {
+      const shareValue = await queryContract({
+        contract: poolAddress,
+        chain,
+        data: `{"share":{"amount":"${lpBalance}"}}`,
+      });
+      return shareValue.map((token) => ({
+        amount: token.amount,
+        denom: token.info.native_token.denom,
+      }))
+    }
   },
-];
+  // Add more DEX configurations here as needed
+};
+
+/***
+ * Value of tokens being held in Valence accounts
+ */
+async function getValenceAccountTvl(api, valenceDomain,chain) {
+ 
+  let cache = await getCache('valence','accounts')
+  let allProgramAccounts = cache.allProgramAccounts
+
+  if (!cache.allProgramAccounts) {
+    // fetch all program configs from the registry contract
+    const allProgramConfigs = await queryValencePrograms(VALENCE_ADDRESSES.VALANCE_PROTOCOL_REGISTRY_ADDRESS);
+    // extract all account addresses from the program configs
+    allProgramAccounts = allProgramConfigs.map((program)=>{
+       return Object.values(program.program_config.accounts) 
+    })
+    await setCache('valence','accounts',{allProgramAccounts})
+  }
 
-async function tvl(api) {
-  for (const pooler of poolerAddresses) {
+  // filter accounts by chain
+  const allAccountAddressesForChain = allProgramAccounts.map((accounts)=>extractAccountsFromProgramConfig(accounts, valenceDomain, chain)).flat();
+  
+  return sumTokens({
+      api,
+      owners: allAccountAddressesForChain,
+      chain
+      })
+}
+
+/***
+ * TVL for liquidity pools deployed with Valence Programs
+ */
+async function getLpTvl(api, chain, dexNames=[]) {
+  const balances = api.getBalances();
+
+  for (const dexName of dexNames) {
+    const dexConfig = DEX_CONFIGS[dexName];
+    if (!dexConfig) {
+      console.log(`Warning: skipping unsupported DEX: ${dexName}`);
+      continue;
+    }
+
+    const possibleLpTokens = Object.keys(balances).filter((address) => address.match(dexConfig.regex));
+
+    possibleLpTokens.forEach(async(lpToken)=>{
+         // regex method does not guarantee an LP token is valid, so handle the error gracefully and ignore the asset in the valuation if this is the case
+         try {
+          const lpBalance = balances[lpToken];
```

Comment:
where can I see these LP tokens on your app UI?

Reply:
@waynebruce0x  Most Valence programs so far do not deploy liquidity, but we do have a few programs that do. 

[here](https://app.valence.zone/programs/119?queryConfig={%22main%22:{%22chainId%22:%22neutron-1%22,%22registryAddress%22:%22neutron1d8me7p72yq95sqnq5jpk34nn4t2vdl30yff29r05250ef92mr80saqcl2f%22,%22rpc%22:%22https://rpc-voidara.neutron-1.neutron.org%22,%22domainName%22:%22neutron%22,%22chainName%22:%22neutron%22},%22external%22:[]}) is an example of one program that holds LP shares in an account.

We only have programs deploying to Astroport for the time being, but if there is meaningful use of other DEX's in the future, we will make a PR to capture those